debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java

```diff
@@ -110,7 +132,24 @@ public static ConnectionFactory patternBasedFactory(String urlPattern, Field...
             String url = findAndReplace(urlPattern, props, varsWithDefaults);
             LOGGER.trace("Props: {}", props);
             LOGGER.trace("URL: {}", url);
-            Connection conn = DriverManager.getConnection(url, props);
+            
+            Connection conn = null;
+            ClassLoader driverClassLoader = classloader;
+            String driverClassName = config.getString(JdbcConfiguration.DRIVER);
+            if (driverClassName == null) {
+                conn = DriverManager.getConnection(url, props);
+            } else {
+                try {
+                    if (driverClassLoader == null) {
+                        driverClassLoader = JdbcConnection.class.getClassLoader();
+                    }
+                    Class<java.sql.Driver> driverClazz = (Class<java.sql.Driver>) Class.forName(driverClassName, true, driverClassLoader);
+                    java.sql.Driver driver = driverClazz.newInstance();
+                    conn = driver.connect(url, props);
+                } catch (ClassNotFoundException|IllegalAccessException|InstantiationException e) {
+                    throw new SQLException(e);
+                }                
+            }
```

Comment:
Do you *need* to pass in the name of the driver class? Seems like you want to provide your own classloader, so if you're doing that passing in a specific name of a driver class isn't that much harder. But maybe I don't understand the mechanics of how you're configuring and initializing the connector.